### PR TITLE
4654 - Remove petition modal

### PR DESCRIPTION
**Pull request by matthopson**
Screen Shot 2020-08-04 at 9 14 11 AM
Screen Shot 2020-08-04 at 9 14 17 AM

### CODE REVIEW

File: web-client/src/presenter/actions/removePetitionFromFormDocumentsAction.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { state } from 'cerebral';
+
+/**
+ * removes the petition document from the form.documents array for replacement
+ *
+ * @param {object} providers the providers object
+ * @param {Function} providers.get the cerebral get function
+ * @param {object} store the cerebral store object
```
Comment (matthopson) [inline web-client/src/presenter/actions/removePetitionFromFormDocumentsAction.js:8]: I need to change this to `providers.store` - I'll do it in another PR if there are no requested changes here.

File: web-client/src/views/PetitionQc/PetitionQcDocumentPreview.jsx [hunk 2]
```diff
@@ -37,13 +41,21 @@ export const PetitionQcDocumentPreview = connect(
...
+          )}
           <div className="padding-top-2">
-            {!isPetitionFile && pdfPreviewUrl && (
+            {pdfPreviewUrl && (
               <Button
                 link
                 className="red-warning push-right"
                 onClick={() => {
-                  openConfirmDeletePDFModalSequence();
+                  if (isPetitionFile) {
```
Comment (rachelschneiderman) [inline web-client/src/views/PetitionQc/PetitionQcDocumentPreview.jsx:54]: I like this refactor a lot!

File: web-client/src/views/ConfirmReplacePetitionModal.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  {
+    cancelSequence: sequences.dismissModalSequence,
+    confirmSequence: sequences[props.confirmSequence],
+  },
+  function ConfirmReplacePetitionModal({ cancelSequence, confirmSequence }) {
+    return (
+      <ModalDialog
+        cancelLabel="No, Keep Current PDF"
+        cancelSequence={cancelSequence}
+        className=""
```
Comment (rachelschneiderman) [inline web-client/src/views/ConfirmReplacePetitionModal.jsx:16]: Nit: can this be removed?

File: web-client/src/presenter/actions/removePetitionFromFormDocumentsAction.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * @returns {void} sets the new state for form.documents
+ */
+
+export const removePetitionFromFormDocumentsAction = ({ get, store }) => {
+  const documents = get(state.form.documents);
+
+  documents.some((document, idx) => {
+    if (document.documentType === 'Petition') {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/removePetitionFromFormDocumentsAction.js:16]: Nit: I believe there's a constant for this?

**Review by rachelschneiderman — APPROVED** [review]

File: web-client/src/views/ConfirmReplacePetitionModal.jsx (hunk 3 above)
Comment (matthopson) [reply web-client/src/views/ConfirmReplacePetitionModal.jsx:16]: Yup.

**Comment by matthopson** [comment]
Thanks, I'll throw these nits on a follow-up to keep the builds down.

**Pull request merged**